src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -3655,6 +3670,58 @@ bool Lowering::TryLowerNegToMulLongOp(GenTreeOp* op, GenTree** next)
     *next = outOp;
     return true;
 }
+
+//----------------------------------------------------------------------------------------------
+// TryContainingCselOp: Attempt contain a condition select node in a parent node.
+//
+// Arguments:
+//    parentNode   - The node to attempt an optimisation on.
+//    childNode   - The child conditional select node that we try to contain.
+//
+// Return Value:
+//    true if the conditional select can be contained
+//
+bool Lowering::TryContainingCselOp(GenTreeHWIntrinsic* parentNode, GenTreeHWIntrinsic* childNode)
+{
+    assert(childNode->GetHWIntrinsicId() == NI_Sve_ConditionalSelect);
+
+    bool canContain = false;
+
+    var_types simdBaseType = parentNode->GetSimdBaseType();
+    if (childNode->Op(3)->IsVectorZero())
+    {
+        switch (parentNode->GetHWIntrinsicId())
+        {
+            case NI_Sve_AddAcross:
```

Comment:
Remind me but why we cannot support these?
- AndAcross
- signed/unsigned MinAcross
- signed MaxAcross

Reply:
Inactive elements are treated as zero so absorbing the mask may yield an incorrect result.

Reply:
ok, but they can be optimized if `mask == AllBitsOne` for example?

Reply:
A quick update, we optimise `Csel(AllTrue, op1, op2) => op1` already [here](https://github.com/dotnet/runtime/blob/6aaadea3a888defc631456dc394c95727a876f31/src/coreclr/jit/lowerarmarch.cpp#L4165-L4207) so don't need to do it for XAcross scenarios.